File: fc-chat/demo/part3/ch3/05-1-message-system-add-channel/src/main/java/net/prostars/messagesystem/dto/websocket/inbound/FetchChannelInviteCodeRequest.java
package net.prostars.messagesystem.dto.websocket.inbound;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import net.prostars.messagesystem.constant.MessageType;
import net.prostars.messagesystem.dto.domain.ChannelId;

public class FetchChannelInviteCodeRequest extends BaseRequest {

  private final ChannelId channelId;

  @JsonCreator
  public FetchChannelInviteCodeRequest(@JsonProperty("channelId") ChannelId channelId) {
    super(MessageType.FETCH_CHANNEL_INVITECODE_REQUEST);
    this.channelId = channelId;
  }

  public ChannelId getChannelId() {
    return channelId;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/inbound/FetchChannelInviteCodeRequest.kt
package com.yscorp.withpush.messagesystem.dto.websocket.inbound

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.yscorp.withpush.messagesystem.constant.MessageType
import com.yscorp.withpush.messagesystem.dto.domain.ChannelId

class FetchChannelInviteCodeRequest @JsonCreator constructor(@JsonProperty("channelId") val channelId: ChannelId) :
    BaseRequest(MessageType.FETCH_CHANNEL_INVITECODE_REQUEST) {

}
